src/os/source/vectorsource.js

```diff
@@ -1683,6 +1683,22 @@ os.source.Vector.prototype.getFilteredFeatures = function(opt_allTime) {
       this.timeModel.intersection(range, true);
     }
 
+    // Hide annotations if needed
+    if (this instanceof plugin.file.kml.KMLSource) {
```

Comment:
Generally speaking, if you only want code to execute in a subclass, you should do so in a function override in that class. This technically creates a circular dependency between `os.source.Vector` and `plugin.file.kml.KMLSource`, though JavaScript lets you get away with this.

This also references plugin code (`plugin.*`) in core application code (`os.*`), which should be avoided. Plugins should be removable without impacting application code.